### ISSUE TITLE
Use the `width` property of `ColumnDef` to size columns instead of jQuery

### ISSUE THREAD

**Issue by erfangc** [erfangc/GigaGrid#52]
@caguthrie your recent upgrade from table to DIV has made a lot of behaviors possible

to give ourselves more performant table do you think it is a good idea to leverage the `width` property of `ColumnDef` instead of assigning a dynamic CSS, which requires manual jQuery action?
# The way it would work
- we allow users to specify exactly how width each column ought to be via `size` on `ColumnDef`
- if no size is specified it is set to `w%` where `w = 1 / visibleColumns.length`
- turn `overflow` on the cells to hidden, to ensure alignment across rows
## Theoretical justification

A column's width is a state that dictates how wide each column is. We already have this `width` property in GigaGrid.

This view fits into the prop/state driven view of React ... the `render()` function of the main component under this proposal would still be a pure function of its `props/state` and we can render everything in 1 cycle - thus improving performance

If we ever need to reason about how wide columns ought to be, we avoid having to deal with direct DOM manipulation in our logic
# How do we see hidden content

It is my belief that if a user wants to see overflowing content in a cell, they will intuitively want to drag a handle

At which point, a drag action on cell borders is an action to change its width ... so we dispatch that action, and the entire app re-renders ... all very 'React-isc' without resorting to complicated lifecycle methods and direct DOM manipulation

please let me know what you think!

**Comment by erfangc** [comment]
@caguthrie I've begun working on this ... in addition to creating type definitions (found a easy way) ... this however, will screw up some scrolling code you've put in ... and may need your advice 

concurrent with this update will be the ability to resize columns using a drag handle in the headers

**Comment by erfangc** [comment]
memo ... WIP, the following CSS changes will allow us to create a single scrollbar for vertical scrolling ... simantenously fixing a problem introduced by the move to explicit column width

# tl;dr

- `giga-grid-right-data-container`: get rid of **float: left**
- `giga-grid-right-data-container` && `giga-grid-left-headers-container`: add **display: inline-block**
- `giga-grid-body-view-port`: add **white-space: nowrap; max-width: 100%; overflow-x: hidden**
- move infinite scrolling `div.placeholder` to be the direct child of `.giga-grid-body-viewport` instead as a row
- `giga-grid-right-data-container`: needs an explicitly set `max-width` = `body-viewport.width - left-headers-container.width` otherwise the horizontal scrollbar won't show up

# longer explanation

`div`s shits the bed when parent (i.e. viewport) width < sum of columns (i.e. children) width, especially if the static/data divs are side-by-side by the magic of **floated: left/right**

This is because `div`s don't naturally come with a width ... and takes the width of their children ... so typically the main data canvas have width ` = sum(children_width)` ... but if that width is > the container element .. then it will wrap ... which is not what we want

making the static/data `div`s both `inline-block` + making their parent `whitespace:no-wrap` fixes the problem

However, the main data container: `giga-grid-right-data-container`: needs an explicitly set `max-width` = `body-viewport.width - left-headers-container.width` 

Otherwise the horizontal scrollbar won't show up ... because `right-data-container` _**is**_ the element that is overflowing `body-viewport` ... so we need to limit it's width

The final innovation here is moving the infinitely scroll "placeholder" `div` outside of the data rows ... and into one layer immediately after `giga-grid-body-viewport`. We couldn't do this before b/c of strictness imposed by `<table/>`s but with `div` `span` we can do whatever we want ...

This solution is great b/c we no longer need to adjust 2 scroll tops when scrolling vertically, the vertical scrollbar that presents itself scrolls both regions

the horizontal scrollbar would still need to sync scroll the _**header**_ though ... but that is WAY more manageable as horizontal scrolling does not require infinite scrolling capability

### elevating placeholders up from among the rows to just below the main viewport
![screen shot 2017-03-11 at 4 33 43 pm](https://cloud.githubusercontent.com/assets/6001694/23827089/98d1b8ce-0679-11e7-84b1-22caeb0757e1.png)

### style changes to the main data container aka `right-data-container` (note the max-width: 100% here is wrong ... we need an explicit number
![screen shot 2017-03-11 at 4 33 51 pm](https://cloud.githubusercontent.com/assets/6001694/23827090/98d1ec40-0679-11e7-9706-25fc2d84f915.png)
